docs/proposals/20230407-managed-k8s-capi-contract-changes.md

```diff
@@ -0,0 +1,380 @@
+---
+title: Contract Changes to Support Managed Kubernetes
+authors:
+  - "@jackfrancis"
+reviewers:
+  - "@richardcase"
+  - "@pydctw"
+  - "@mtougeron"
+  - "@CecileRobertMichon"
+  - "@fabriziopandini"
+  - "@sbueringer"
+  - "@killianmuldoon"
+  - "@mboersma"
+  - "@nojnhuh"
+creation-date: 2023-04-07
+last-updated: 2023-04-07
+status: provisional
+see-also:
+  - "/docs/proposals/20220725-managed-kubernetes.md"
+---
+
+# Contract Changes to Support Managed Kubernetes
```

Comment:
Should this be retitled a bit to focus only on the cluster endpoints? It seems the proposal is mostly focused on that + the optionality of the InfraCluster resource, which haven't fully grasped which issue is trying to solve 😄

Reply:
We had a thread about this here:

- https://github.com/kubernetes-sigs/cluster-api/pull/8500#pullrequestreview-1405801656

(happy to revive btw, changing a title is easy :))

Reply:
@vincepri WDYT about this now after the latest changes?